site/zenodo_rdm/legacy/requests/record_upgrade.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo legacy record upgrade request."""
+
+from invenio_access.permissions import system_identity
+from invenio_i18n import gettext as _
+from invenio_rdm_records.proxies import current_record_communities_service
+from invenio_records_resources.services.uow import IndexRefreshOp, RecordCommitOp
+from invenio_requests.customizations import CommentEventType, RequestType, actions
+from invenio_requests.proxies import current_events_service, current_requests_service
+
+
+def _remove_permission_flag(record):
+    """Remove can_community_read_files permission flag."""
+    try:
+        record.parent.permission_flags.pop("can_community_read_files")
+        record.parent.commit()
+    except (KeyError, AttributeError):
+        # if field or flag is absent just continue with an action
+        pass
+
+
+def _remove_record_from_communities(record, identity):
+    """Remove record from all the communities."""
+    communities_ids = record.parent.communities.ids
+    communities = []
+    for community_id in communities_ids:
+        communities.append({"id": community_id})
+
+    data = dict(communities=communities)
+    current_record_communities_service.remove(identity, record["id"], data)
+
+
+#
+# Actions
+#
+class SubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        # Set the record's title as the request title.
+        self.request["title"] = record.metadata["title"]
```

Comment:
We are missing to have an extra msg here, with a nice explanation of what is this request about.
Ping @slint for suggestions if he wants to have this in Zenodo.

Reply:
Something like `Upgrade your legacy record {record.metadata.title}` would also be fine. I see that this is the "title" of the description below, so we can skip it below then.

Reply:
@slint You mean only leave the title  `Upgrade your legacy record {record.metadata.title}` without any explanation in the description? 
The thing is, this description is the only place where we explain to the user what is happening any why they have to complete this request, so I would leave it.